test/server/options_impl_test.cc

```diff
@@ -60,12 +60,32 @@ TEST(OptionsImplTest, InvalidCommandLine) {
                           "Couldn't find match for argument");
 }
 
+TEST(OptionsImplTest, v2True) {
+  std::unique_ptr<OptionsImpl> options = createOptionsImpl(
+      "envoy --mode validate --concurrency 2 -c hello --admin-address-path path --restart-epoch 1 "
+      "--local-address-ip-version v6 -l info --service-cluster cluster --service-node node "
+      "--service-zone zone --file-flush-interval-msec 9000 --drain-time-s 60 --log-format [%v] "
+      "--parent-shutdown-time-s 90 --log-path /foo/bar --v2-config-only 1 --disable-hot-restart");
```

Comment:
tclap only supports 0/1 rather than false/true? We might want to use an independent flag to avoid breaking folks production equivalent of Borg config.

Reply:
Yeah, I tested and true/false chokes.  I was tempted to do a string literal true/false but I figured folks use to TCLAP rather than google flags would find that more confusing

Reply:
Also I'm open to just adding a second flag but if the concern is folks who want v2 only will have it specified, we're going to break them eventually when we remove v1 config and the flag.  Is it worth adding the complexity of a second flag to avoid a problem they're going to face in a few months anyway?

Reply:
IMO we should add a second flag and not change this one (basically make it do nothing since the default is already what it implies). I think this is less likely to break people?